packages/wallet/src/redux/actions.ts

```diff
@@ -111,15 +129,15 @@ export interface FundingReceivedEvent {
 }
 
 export interface ChallengeExpiredEvent {
-  type: "WALLET.ADJUDICATOR.CHALLENGE_EXPIRED";
+  type: WalletActionType.WALLET_ADJUDICATOR_CHALLENGE_EXPIRED;
   processId: string;
   protocolLocator: ProtocolLocator;
   channelId: string;
   timestamp: number;
 }
 
 export interface ChannelUpdate {
-  type: "WALLET.ADJUDICATOR.CHANNEL_UPDATE";
+  type: WalletActionType.WALLET_ADJUDICATOR_CHANNEL_UPDATE;
```

Comment:
The type of `type` in this interface is not `string`. It is `"WALLET.ADJUDICATOR.CHANNEL_UPDATE"`. Therefore, `"WALLET.ADJUDICATOR.CHANNEL_UPDATE" ` is the only allowable value of `o.type` for an object `o` of type `ChannelUpdate`.

As an example, Typescript uses this fact in a union type, and infers that when `o` is of type `ChannelUpdate | AdjudicatorKnown`, then `o.type` is of type `"WALLET.ADJUDICATOR.CHANNEL_UPDATE" | "WALLET.ADJUDICATOR_KNOWN"`. An editor like VSCode can use this type information to provide autocomplete, for instance, and to display type errors in such union types.

I'm not aware of any additional type safety introduced by replacing the type of `type` with an enum. Is there something we're unaware of?

Reply:
The type of `type` is not a string but it is a using a string value to represent the type. The rationale for this change was to introduce a common reference for this string value that's being used to define the type of an object, such that those who don't get these inferences (and therefore, autocomplete) have a common reference and are less prone to running into build issues.

But all of this seems like it's coming at too large of an expense, so I digress and we can close this.